instrumentation/netty/netty-4.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/Netty41ClientAgentTest.java

```diff
@@ -0,0 +1,311 @@
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTestServer;
+import java.util.NoSuchElementException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class Netty41ClientAgentTest {
+
+  private HttpClientTestServer server;
+
+  @RegisterExtension
+  static InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @BeforeEach
+  void setUp() {
+    server = new HttpClientTestServer(testing.getOpenTelemetry());
+    server.start();
+  }
+
+  @AfterEach
+  void tearDown() {
+    server.stop();
+  }
+
+  @Test
+  @DisplayName("test connection reuse and second request with lazy execute")
+  void testConnectionReuse() throws InterruptedException {
+    // Create a simple Netty pipeline
+    EventLoopGroup group = new NioEventLoopGroup();
+    Bootstrap b = new Bootstrap();
+    b.group(group)
+        .channel(NioSocketChannel.class)
+        .handler(
+            new ChannelInitializer<SocketChannel>() {
+              @Override
+              protected void initChannel(SocketChannel socketChannel) throws Exception {
+                ChannelPipeline pipeline = socketChannel.pipeline();
+                pipeline.addLast(new HttpClientCodec());
+              }
+            });
+    DefaultFullHttpRequest request =
+        new DefaultFullHttpRequest(
+            HttpVersion.HTTP_1_1, HttpMethod.GET, "/success", Unpooled.EMPTY_BUFFER);
+    request.headers().set(HttpHeaderNames.HOST, "localhost:" + server.httpPort());
+    // note that this is a purely asynchronous request
+    Channel ch =
+        testing.runWithSpan(
+            "parent1",
+            () -> {
+              Channel channel = b.connect("localhost", server.httpPort()).sync().channel();
+              channel.write(request);
+              channel.flush();
+              return channel;
+            });
+    // This is a cheap/easy way to block/ensure that the first request has finished and check
+    // reported spans midway through
+    // the complex sequence of events
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasNoParent().hasName("parent1").hasKind(SpanKind.INTERNAL),
+                span -> span.hasParent(trace.getSpan(0)).hasKind(SpanKind.CLIENT),
+                span -> span.hasKind(SpanKind.SERVER).hasParent(trace.getSpan(1))));
+
+    // now run a second request through the same channel
```

Comment:
perhaps it would be better to call `testing.clearData()` before the second request so that we wouldn't need to assert the first request again

Reply:
done